### PR TITLE
Added encrypted_transport testcase migrated in go

### PR DESCRIPTION
**Pull request by saurabh408**
Encrypted transport tests :
   - Sets up all the certificates , some table ACL's and connect via grpc client with SSL.
   - exercises the "grpc_use_effective_callerid" flag, by connecting without SSL.
 
Signed-off-by: saurabh <saurabh.408@gmail.com>

### CODE REVIEW

File: go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+	assert.Nil(t, err)
+
+	// create database so vttablet can start behaving normally
+	err = masterTablet.VttabletProcess.CreateDB(keyspace)
+	assert.Nil(t, err)
+
+	err = clusterInstance.VtctlclientProcess.InitTablet(&replicaTablet, clusterInstance.Cell, keyspace, hostname, shardName)
+	assert.Nil(t, err)
+	err = replicaTablet.VttabletProcess.CreateDB(keyspace)
+	assert.Nil(t, err)
```
Comment (ajeetj) [inline go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go]: Maybe use in a loop
```
for _, tablet := range []cluster.Vttablet{*masterTablet, *replicaTablet} {
			
		}
```

File: go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+
+	_, err = f.WriteString(`{
+	"table_groups": [
+	{
+		"table_names_or_prefixes": ["vt_insert_test"],
+		"readers": ["vtgate client 1"],
+		"writers": ["vtgate client 1"],
+		"admins": ["vtgate client 1"]
+	}
+	]
```
Comment (ajeetj) [inline go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go]: nit: formatting looks off

File: go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+	}
+
+	// setup replication
+	var vtctlClientArgs []string
+
+	vtctlClientTmArgs := append(vtctlClientArgs, tmclientExtraArgs("vttablet-client-1")...)
+
+	// Reparenting
+	vtctlClientArgs = append(vtctlClientTmArgs, "InitShardMaster", "-force", "test_keyspace/0", masterTablet.Alias)
+	_, err = clusterInstance.VtctlProcess.ExecuteCommandWithOutput(vtctlClientArgs...)
```
Comment (ajeetj) [inline go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go]: As you don't need a result here maybe write another method in VtctlProcess which just returns err?

File: go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+		if err := clusterInstance.VtctlProcess.CreateKeyspace(keyspace.Name); err != nil {
+			return 1, err
+		}
+		shard := &cluster.Shard{
+			Name: shardName,
+		}
+		for i := 0; i < 2; i++ {
+			// instantiate vttablet object with reserved ports
+			tabletUID := clusterInstance.GetAndReserveTabletUID()
+			tablet := clusterInstance.GetVttabletInstance("replica", tabletUID, cell)
```
Comment (ajeetj) [inline go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go]: nit: if you can pass tabletUID=0 then you don't need to call separate `tabletUID := clusterInstance.GetAndReserveTabletUID()`

File: go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go (hunk 1 above)
Comment (saurabh408) [reply go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go]: done

File: go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go (hunk 2 above)
Comment (saurabh408) [reply go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go]: done

File: go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go (hunk 3 above)
Comment (saurabh408) [reply go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go]: done

File: go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go (hunk 4 above)
Comment (saurabh408) [reply go/test/endtoend/encryption/encryptedtransport/encrypted_transport_test.go]: done

**Pull request merged**